modelforge/dataset/dataset.py

```diff
@@ -102,217 +103,186 @@ class Metadata:
         return self
 
-
-NNPInputTuple = NamedTuple(
-    "NNPInputTuple",
-    [
-        ("atomic_numbers", torch.Tensor),
-        ("positions", torch.Tensor),
-        ("atomic_subsystem_indices", torch.Tensor),
-        ("total_charge", torch.Tensor),
-        ("pair_list", torch.Tensor),
-        ("partial_charge", torch.Tensor),
-        ("box_vectors", torch.Tensor),
-        ("is_periodic", torch.Tensor),
-    ],
-)
+    def to_dtype(self, dtype: torch.dtype):
+        self.per_system_energy = self.per_system_energy.to(dtype)
+        self.per_atom_force = self.per_atom_force.to(dtype)
+        self.per_system_dipole_moment = self.per_system_dipole_moment.to(dtype)
+        return self
 
 
-@dataclass
+# Define the input class
 class NNPInput:
-    """
-    A dataclass to structure the inputs for neural network potentials.
-
-    Attributes
-    ----------
-    atomic_numbers : torch.Tensor
-        A 1D tensor containing atomic numbers for each atom in the system(s).
-        Shape: [num_atoms], where `num_atoms` is the total number of atoms
-        across all systems.
-    positions : torch.Tensor
-        A 2D tensor of shape [num_atoms, 3], representing the XYZ coordinates of
-        each atom.
-    atomic_subsystem_indices : torch.Tensor
-        A 1D tensor mapping each atom to its respective subsystem or molecule.
-        This allows for calculations involving multiple molecules or subsystems
-        within the same batch. Shape: [num_atoms].
-    total_charge : torch.Tensor
-        A tensor with the total charge of molecule. Shape: [num_systems,1], where
-        `num_systems` is the number of molecules.
-    pair_list : Optional[torch.Tensor]
-        Pair list for neighbor interactions.
-    box_vectors : Optional[torch.Tensor]
-        A 2D tensor of shape [3, 3], representing the box vectors in a system.
-        Currently, only supports a single box vector for all systems, as this is primarily meant for
-        inference on a single system, not training.
-    is_periodic : Optional[bool]
-        A boolean indicating whether the system is periodic or not.
-    """
-
-    atomic_numbers: torch.Tensor
-    positions: Union[torch.Tensor, Quantity]
-    atomic_subsystem_indices: torch.Tensor
-    total_charge: torch.Tensor
-    pair_list: Optional[torch.Tensor] = None
-    partial_charge: Optional[torch.Tensor] = None
-    box_vectors: Optional[torch.Tensor] = torch.zeros(3, 3)
-    is_periodic: Optional[torch.Tensor] = torch.Tensor([False])
+    __slots__ = (
+        "atomic_numbers",
+        "positions",
+        "atomic_subsystem_indices",
+        "total_charge",
+        "pair_list",
+        "partial_charge",
+        "box_vectors",
+        "is_periodic",
+    )
 
-    def to(
+    def __init__(
         self,
-        *,
-        device: Optional[torch.device] = None,
-        dtype: Optional[torch.dtype] = None,
+        atomic_numbers: torch.Tensor,
+        positions: torch.Tensor,
```

Comment:
This now drops the option for positions to have units.  I think that is probably fine, but we'll probably want to make a helper function that returns an NNPInput instance,  where positions and box vectors require units to be attached.

Reply:
You mean a second implementation with units?